src/utils.ts

```diff
@@ -1767,94 +2094,164 @@ export class newText extends newShape {
       multiLine = false,
       font = 'sans-serif',
       align = 'left',
-      baseline = 'alphabetic',
+      baseline = 'top',
       style = '',
       orientation = 0,
       color = "hsl(0, 0%, 0%)",
-      backgroundColor = "hsla(0, 0%, 100%, 0)"
+      backgroundColor = "hsla(0, 0%, 100%, 0)",
+      scale = new Vertex(1, 1)
     }: TextParams = {}) {
-      super();
-      this.boundingBox = new newRect(origin, new Vertex(width, height));
-      this.boundingBox.fillStyle = backgroundColor;
-      this.boundingBox.strokeStyle = backgroundColor;
-      this.fontsize = fontsize;
-      this.multiLine = multiLine;
-      this.font = font;
-      this.align = align;
-      this.baseline = baseline;
-      this.style = style;
-      this.orientation = orientation;
-      this.fillStyle = color;
-      this.words = this.getWords();
-    }
+    super();
+    this.boundingBox = new newRect(origin, new Vertex(width, height));
+    this.boundingBox.fillStyle = backgroundColor;
+    this.boundingBox.strokeStyle = backgroundColor;
+    this.boundingBox.lineWidth = 1e-6; //TODO: this is a HOT FIX
+    this.fontsize = fontsize;
+    this.multiLine = multiLine;
+    this.font = font;
+    this.style = style;
+    this.orientation = orientation;
+    this.fillStyle = color;
+    this.words = this.getWords();
+    this.align = align;
+    this.baseline = baseline;
+    this.scale = scale;
+  }
+
+  private static buildFont(style: string, fontsize: number, font: string): string { return `${style} ${fontsize}px ${font}` }
 
-  private static buildFont(style: string, fontsize: number, font: string): string {
-    return `${style} ${fontsize}px ${font}`
+  public static deserializeTextParams(data: any): TextParams {
+    const style = `${data.text_style?.bold ? "bold" : ""}${data.text_style?.bold || data.text_style?.italic ? " " : ""}${data.text_style?.italic ? "italic" : ""}`;
+    return {
+      width: data.max_width,
+      height: data.height,
+      fontsize: data.text_style?.font_size,
+      multiLine: data.multi_lines,
+      font: data.text_style?.font,
+      align: data.text_style?.text_align_x,
+      baseline: data.text_style?.text_align_y,
+      style: style,
+      orientation: data.text_style?.angle,
+      color: data.text_style?.text_color
+    } as TextParams
+  }
+
+  public static deserialize(data: any, scale: Vertex): newText {
+    const textParams = newText.deserializeTextParams(data);
+    const text = new newText(data.comment, new Vertex(data.position_x, data.position_y), textParams);
+    text.isScaled = data.text_scaling ?? false;
+    text.scale = new Vertex(scale.x, scale.y);
+    return text
   }
 
   get fullFont(): string { return newText.buildFont(this.style, this.fontsize, this.font) }
 
+  private getCoonsUnscaled(): [Vertex, Vertex] {
```

Comment:
What is a coon ? Because the definition I have is quite peculiar and pejorative (to say the least) ^^'

Reply:
Thanks, i wanted to say Corner ><'